Tests/Mapper/BaseGroupedMapperTest.php

```diff
@@ -152,6 +166,32 @@ public function testEndException()
         $this->baseGroupedMapper->end();
     }
 
+    public function labelDataProvider()
+    {
+        return array(
+            array('fooGroup1', null, 'label_foogroup1'),
+            array('fooGroup1', 'custom_label', 'custom_label'),
+            array('form.label_foo', null, 'form.label_foo'), // BC layer
```

Comment:
If you can, specify keys describing each dataset : 

```php
array(
    'nominal use case' => array('fooGroup1', null, 'label_foogroup1'),
    'custom label' =>  array('fooGroup1', 'custom_label', 'custom_label'),
    'legacy call' => array('form.label_foo', null, 'form.label_foo'),
)
```
This should make error messages a bit clearer.